translib/db/db_table_pattern_test.go

```diff
@@ -0,0 +1,223 @@
+//  See the License for the specific language governing permissions and       //
+//  limitations under the License.                                            //
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+package db
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+var allKeysPat = Key{Comp: []string{"*"}}
+var singleKeyPat = Key{Comp: []string{"KEY1"}}
+var genericKeyPat = Key{Comp: []string{"KEY*"}}
+
+func BenchmarkGetTablePattern(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		if _, e := db.GetTablePattern(&ts, allKeysPat); e != nil {
+			b.Errorf("GetTablePattern() returns err: %v", e)
+		}
+	}
+}
+
+func BenchmarkGetTableOrig(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		if _, e := db.GetTable(&ts); e != nil {
+			b.Errorf("GetTable() returns err: %v", e)
+		}
+	}
+}
+
+func TestGetTablePattern(t *testing.T) {
+	if _, e := db.GetTablePattern(&ts, allKeysPat); e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+}
+
+func TestGetTablePatternSingular(t *testing.T) {
+	if _, e := db.GetTablePattern(&ts, singleKeyPat); e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+}
+
+func TestGetTablePatternGeneric(t *testing.T) {
+	if _, e := db.GetTablePattern(&ts, genericKeyPat); e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+}
+
+func TestGetTablePatternEmpty(t *testing.T) {
+	if _, e := db.GetTablePattern(&TableSpec{Name: "UNLIKELY_23"}, allKeysPat); e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+}
+
+func TestGetTableOrig(t *testing.T) {
+	if _, e := db.GetTable(&ts); e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+}
+
+func TestGetTableOrigEmpty(t *testing.T) {
+	if _, e := db.GetTable(&TableSpec{Name: "UNLIKELY_23"}); e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+}
+
+func TestGetTablePatternCompOrig(t *testing.T) {
+	tPat, e := db.GetTablePattern(&ts, allKeysPat)
+	if e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+
+	tOrig, e := db.GetTable(&ts)
+	if e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+
+	// The ordering of the arrays in patterns map values matters
+	delete(tPat.patterns, db.key2redis(&ts, Key{Comp: []string{"*"}}))
+	delete(tOrig.patterns, db.key2redis(&ts, Key{Comp: []string{"*"}}))
+
+	if !reflect.DeepEqual(tPat, tOrig) {
+		fmt.Println("\ntPat: \n", tPat)
+		fmt.Println("\ntOrig: \n", tOrig)
+		t.Errorf("GetTable() != GetTablePattern")
+	}
+}
+
+func TestGetTablePatternCompOrigEmpty(t *testing.T) {
+	tsEmpty := TableSpec{Name: "UNLIKELY_23"}
+	tPat, e := db.GetTablePattern(&tsEmpty, allKeysPat)
+	if e != nil {
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+
+	tOrig, e := db.GetTable(&tsEmpty)
+	if e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+
+	if !reflect.DeepEqual(tPat, tOrig) {
+		fmt.Println("\ntPat: \n", tPat)
```

Comment:
Same. Suggest to replace fmt.Println() to  t.Logf()

Reply:
A review has been raised for fixing the issues raised in the review comments: 
https://github.com/sonic-net/sonic-mgmt-common/pull/106